packages/client/src/util/environment.ts

```diff
@@ -88,24 +88,27 @@ function getGlobalFallbackBranch(): string | undefined {
 }
 
 export async function getGitBranch(): Promise<string | undefined> {
+  const cmd = ['git', 'branch', '--show-current'];
+
   // Node.js: child_process.execSync
   try {
+    // CJS
     if (typeof require === 'function') {
       const req = require; // Avoid "Detected a Node builtin module import while Node compatibility is disabled" in CloudFlare Workers
-      return req('child_process').execSync('git branch --show-current', { encoding: 'utf-8' }).trim();
+      return req('child_process').execSync(cmd.join(' '), { encoding: 'utf-8' }).trim();
     }
+
+    // ESM
+    const { execSync } = await import('child_process');
```

Comment:
I wonder if this will trigger warnings on wrangler during its compile time like require did.

Maybe we will need a, although not sure if that will work either...

```
const imp = import
const { execSync } = await imp('child_process');
```

Reply:
Tested and 🥁 

![image](https://user-images.githubusercontent.com/2181866/180272591-18ef2f89-5a19-42e8-bb4c-8151c8f45385.png)

We do need some trick for `import(foo)`

Reply:
Great catch! Should be fixed with this https://github.com/xataio/client-ts/pull/474/commits/22e351ca36914df0e618c697b762529d566776d2